provisioning/miniprov/hare_mp/main.py

```diff
@@ -569,6 +569,8 @@ def init(args):
         # Stopping hax and consul
         if is_mkfs_required(url):
             stop_hax_blocking(hax_starter)
+        else:
```

Comment:
Randomly sleeping is very bad as it can be affected with slight change in the other parameters. Its better to have loop around some condition that is met eventually.

Reply:
I think following is making sure that most of the pre-part is sync-up, 
        while not is_mkfs_done_on_all_nodes(utils, cns_utils,
                                            data_nodes):
            sleep(5)
            
I can think of one more part which can help it make robust for post  is make sure all the motr ioservices are online.